sdl-core/src/main/scala/io/smartdatalake/workflow/action/generic/transformer/SQLDfsTransformer.scala

```diff
@@ -22,47 +22,72 @@ package io.smartdatalake.workflow.action.generic.transformer
  * @param description    Optional description of the transformer
- * @param code           SQL code for transformation.
+ * @param code           Map of output names and corresponding SQL code for transformation.
+ *                       If this is the last transformation in the chain, the output name has to match an output DataObject id,
+ *                       otherwise it can be any name which will then be available in the next transformation.
  *                       Use tokens %{<key>} to replace with runtimeOptions in SQL code.
  *                       Example: "select * from test where run = %{runId}"
- *                       A special token %{inputViewName} can be used to insert the temporary view name.
+ *                       The special token ${inputViewName_<input_name>} can be used to insert the name of temporary views.
+ *                       The input name is either the id of an input DataObject, or the name of an output of the previous transformation
+ *                       if this is not the first transformation of the chain.
  * @param options        Options to pass to the transformation
  * @param runtimeOptions optional tuples of [key, spark sql expression] to be added as additional options when executing transformation.
  *                       The spark sql expressions are evaluated against an instance of [[DefaultExpressionData]].
  */
-case class SQLDfsTransformer(override val name: String = "sqlTransform", override val description: Option[String] = None, code: Map[DataObjectId,String], options: Map[String, String] = Map(), runtimeOptions: Map[String, String] = Map()) extends OptionsGenericDfsTransformer {
+case class SQLDfsTransformer(override val name: String = "sqlTransform", override val description: Option[String] = None, code: Map[String,String], options: Map[String, String] = Map(), runtimeOptions: Map[String, String] = Map())
+  extends OptionsGenericDfsTransformer with SmartDataLakeLogger {
   override def transformWithOptions(actionId: ActionId, partitionValues: Seq[PartitionValues], dfs: Map[String,GenericDataFrame], options: Map[String, String])(implicit context: ActionPipelineContext): Map[String,GenericDataFrame] = {
     val functions = DataFrameSubFeed.getFunctions(dfs.values.head.subFeedType)
-    // register all input DataObjects as temporary table
-    dfs.foreach {
-      case (dataObjectId,df) =>
-        val objectId =  ActionHelper.replaceSpecialCharactersWithUnderscore(dataObjectId)
-        // Using createTempView does not work because the same data object might be created more than once
-        df.createOrReplaceTempView(objectId)
+    // register all inputs as temporary table
+    val inputViewNameOptions = dfs.map {
+      case (inputName,df) =>
+        val inputViewName =  ActionHelper.createTemporaryViewName(inputName)
+        // Using createTempView does not work because the same temporary view is created more than once (Init & Exec phase...)
```

Comment:
Confusing comment since the function createTempView is nowhere to be found, i suggest to remove it